### PR TITLE
ci: increase minimum Bazel version to 4.2.1

### PR DESCRIPTION
**Pull request by coryan**
Also fixed some typos that I found while scanning over the top-level README file.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8883)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -458,13 +459,13 @@ policy described at https://cloud.google.com/terms.
...
 the C++ symbols and headers. They also need to reference the name of the
-library in their build scripts. Depending of the build system they use
+library in their build scripts. Depending on the build system they use
 this may be a CMake target, a Bazel rule, a pkg-config module, or just the
 name of some object in the file system.
 
 As with the C++ API, we try to avoid breaking changes to these interface
 points. Sometimes such changes yield benefits to our customers, in the form of
-easier-to-understand what names go with with services, or more consistency
+easier-to-understand what names go with services, or more consistency
```
Comment (dbolduc) [inline README.md:468]: This clause doesn't really parse.

maybe: "... in the form of bug fixes, easier-to-understand names, or increased consistency across services."

**Review by dbolduc — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (coryan) [reply README.md]: Fixed.

File: README.md [hunk 2]
```diff
@@ -465,9 +465,9 @@ name of some object in the file system.
 
 As with the C++ API, we try to avoid breaking changes to these interface
 points. Sometimes such changes yield benefits to our customers, in the form of
-easier-to-understand what names go with services, or more consistency
-across services. When these benefits warrant it, we will announce these changes
-prominently in our `CHANGELOG.md` file and in the affected release's notes.
+bug fixes, increased consistency across services, or easier to understand names.
+When these benefits warrant it, we will announce these changes prominently in
+our `CHANGELOG.md` file and in the affected release's notes.
```
Comment (dbolduc) [inline README.md:470]: nit: "Nevertheless," fits on this line

Comment (coryan) [reply README.md:470]: And then I would have to reflow the whole paragraph. I was being lazy.

**Comment by coryan** [comment]
Closing.  I will open a separate PR to apply the documentation fixes.

**Pull request closed without merging**